src/components/orders/OrderPriceDisplay/index.tsx

```diff
@@ -47,28 +42,30 @@ export function OrderPriceDisplay(props: Props): JSX.Element {
   } = props
 
   const [isPriceInverted, setIsPriceInverted] = useState(initialInvertedPrice)
-  const invert = (): void => setIsPriceInverted((curr) => !curr)
-
-  const calculatedPrice = calculatePrice({
-    denominator: { amount: buyAmount, decimals: buyToken.decimals },
-    numerator: { amount: sellAmount, decimals: sellToken.decimals },
-  })
-  const displayPrice = (isPriceInverted ? invertPrice(calculatedPrice) : calculatedPrice).toString(10)
-  const formattedPrice = formatSmart({
-    amount: displayPrice,
-    precision: NO_ADJUSTMENT_NEEDED_PRECISION,
-    smallLimit: HIGH_PRECISION_SMALL_LIMIT,
-    decimals: HIGH_PRECISION_DECIMALS,
-  })
-
-  const buySymbol = safeTokenName(buyToken)
-  const sellSymbol = safeTokenName(sellToken)
+  const [formattedPrice, setFormattedPrice] = useState('')
+  const invert = (): void => setIsPriceInverted(!isPriceInverted)
 
-  const [baseSymbol, quoteSymbol] = isPriceInverted ? [sellSymbol, buySymbol] : [buySymbol, sellSymbol]
+  React.useEffect((): void => {
+    const price: string = constructPrice({
```

Comment:
I suggest to separate the price calculation from the price formatting for display purposes.

Also, you could probably use these helper functions to calculate the prices

https://github.com/gnosis/gp-ui/blob/36eae4570aac573e71d909042d3d4b18adfbf5aa/src/utils/operator.ts#L172 

https://github.com/gnosis/gp-ui/blob/36eae4570aac573e71d909042d3d4b18adfbf5aa/src/utils/operator.ts#L195